airbyte-integrations/connectors/source-amazon-ads/unit_tests/test_streams.py

```diff
@@ -0,0 +1,306 @@
+    )
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    campaigns_stream = get_stream_by_name(streams, "sponsored_display_campaigns")
+
+    with pytest.raises(requests.exceptions.HTTPError):
+        get_all_stream_records(campaigns_stream)
+
+
+@responses.activate
+def test_streams_campaigns_pagination_403_error_expected(mocker, test_config, profiles_response, campaigns_response):
+    setup_responses(
+        profiles_response=profiles_response,
+    )
+    responses.add(
+        responses.GET,
+        "https://advertising-api.amazon.com/sd/campaigns",
+        json={"code": "403", "details": "details", "requestId": "xxx"},
+        status=403,
+    )
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    campaigns_stream = get_stream_by_name(streams, "sponsored_display_campaigns")
+
+    campaigns_records = get_all_stream_records(campaigns_stream)
+    assert campaigns_records == []
+
+
+@pytest.mark.parametrize(
+    ("stream_name", "endpoint"),
+    [
+        ("sponsored_display_ad_groups", "sd/adGroups"),
+        ("sponsored_display_product_ads", "sd/productAds"),
+        ("sponsored_display_targetings", "sd/targets"),
+    ],
+)
+@responses.activate
+def test_streams_displays(
+    test_config,
+    stream_name,
+    endpoint,
+    profiles_response,
+    adgroups_response,
+    targeting_response,
+    product_ads_response,
+):
+    setup_responses(
+        profiles_response=profiles_response,
+        adgroups_response=adgroups_response,
+        targeting_response=targeting_response,
+        product_ads_response=product_ads_response,
+    )
+
+    source = SourceAmazonAds()
+    streams = source.streams(test_config)
+    profile_stream = get_stream_by_name(streams, "profiles")
+    test_stream = get_stream_by_name(streams, stream_name)
+    _ = get_all_stream_records(profile_stream)
+
+    records = get_all_stream_records(test_stream)
+    assert len(records) == 4
+    schema = test_stream.get_json_schema()
+    for r in records:
+        validate(schema=schema, instance=r)
+    assert any([endpoint in call.request.url for call in responses.calls])
+
+
+@pytest.mark.parametrize(
+    ("stream_name", "endpoint"),
+    [
+        ("sponsored_brands_campaigns", "sb/campaigns"),
+        ("sponsored_brands_ad_groups", "sb/adGroups"),
+        ("sponsored_brands_keywords", "sb/keywords"),
+        ("sponsored_product_campaigns", "v2/sp/campaigns"),
+        ("sponsored_product_ad_groups", "v2/sp/adGroups"),
+        ("sponsored_product_keywords", "v2/sp/keywords"),
+        ("sponsored_product_negative_keywords", "v2/sp/negativeKeywords"),
+        ("sponsored_product_ads", "v2/sp/productAds"),
+        ("sponsored_product_targetings", "v2/sp/targets"),
+    ],
+)
+@responses.activate
+def test_streams_brands_and_products(
+    test_config,
+    stream_name,
+    endpoint,
+    profiles_response,
+    adgroups_response,
+    targeting_response,
+    product_ads_response,
```

Comment:
These 3 fixtures are not used in this test. Maybe they have to be removed?

Reply:
Done